### ISSUE TITLE
Pytorch Model with multiple inputs

### ISSUE THREAD

**Issue by agemagician** [NVIDIA/apex#58]
There is a bug that doesn't allow a model to have multiple inputs through the forward function after using network_to_half function.
I made a model with 2 inputs parameters, and it works fine without network_to_half. However, after using it, it gives me the following error:
forward() takes 2 positional arguments but 3 were given

**Comment by mcarilli** [comment]
Sorry, it's been a busy last couple of weeks and I overlooked this question.  It appears to be a limitation of `torch.nn.Sequential` which is used to define [network_to_half](https://github.com/NVIDIA/apex/blob/master/apex/fp16_utils/fp16util.py#L41) .  I'll double check with the Pytorch people whether `torch.nn.Sequential` is intended to support multiple inputs.  Seems like a reasonable requirement.  Maybe there's something obvious I'm missing.

In the meantime, what `network_to_half` is doing is really simple:  1. converting input to half if it isn't already, and 2. converting batchnorms to float.  Therefore, as long as you manually call .half() on your inputs before starting the forward pass, you don't need 1., which means you don't need the full `network_to_half` Sequential sequence.  Instead of
```
my_model = network_to_half(my_model)
```
you can implement only 2. by saying
```
my_model = BN_convert_float(my_model.half())
```
and after that `my_model` should still be able to handle whatever input structure you intended.
